mlflow/sklearn/utils.py

```diff
@@ -103,6 +118,248 @@ def _get_args_for_score(score_func, fit_func, fit_args, fit_kwargs):
+    to average outputs with uniform weight.
+
+    Steps:
+    1. Extract X and y_true from fit_args and fit_kwargs, and compute y_pred.
+    2. If the sample_weight argument exists in fit_func (accuracy_score by default
+    has sample_weight), extract it from fit_args or fit_kwargs as
+    (y_true, y_pred, sample_weight, multioutput), otherwise as (y_true, y_pred, multioutput)
+    3. return a dictionary of metric(name, value)
+
+    :param trained_estimator: The already fitted regressor
+    :param fit_args: Positional arguments given to fit_func.
+    :param fit_kwargs: Keyword arguments given to fit_func.
+    :return: dictionary of (function name, computed value)
+    """
+    import sklearn
+
+    fit_arg_names = _get_arg_names(trained_estimator.fit)
+    _, y_true, y_pred = _get_samples_labels_and_predictions(
+        trained_estimator, fit_args, fit_kwargs, fit_arg_names
+    )
+    sample_weight = (
+        _get_sample_weight(fit_arg_names, fit_args, fit_kwargs)
+        if _SAMPLE_WEIGHT in fit_arg_names
+        else None
+    )
+
+    regressor_metrics = [
+        _SklearnMetric(
+            name="mse",
+            function=sklearn.metrics.mean_squared_error,
+            arguments=dict(
+                y_true=y_true,
+                y_pred=y_pred,
+                sample_weight=sample_weight,
+                multioutput="uniform_average",
+            ),
+        ),
+        _SklearnMetric(
+            name="mae",
+            function=sklearn.metrics.mean_absolute_error,
+            arguments=dict(
+                y_true=y_true,
+                y_pred=y_pred,
+                sample_weight=sample_weight,
+                multioutput="uniform_average",
+            ),
+        ),
+        _SklearnMetric(
+            name="r2_score",
+            function=sklearn.metrics.r2_score,
+            arguments=dict(
+                y_true=y_true,
+                y_pred=y_pred,
+                sample_weight=sample_weight,
+                multioutput="uniform_average",
+            ),
+        ),
+    ]
+
+    # To be compatibale with deprecated versions of scikit-learn (below 0.22.2), where
+    # `sklearn.metrics.mean_squared_error` does not have "squared" parameter to calculate `rmse`,
+    # we compute it through np.sqrt(<value of mse>)
+    metrics_value_dict = _get_metrics_value_dict(regressor_metrics)
+    metrics_value_dict["rmse"] = np.sqrt(metrics_value_dict["mse"])
+
+    return metrics_value_dict
+
+
+def _log_warning_for_metrics(func_name, func_call, err):
+    msg = (
+        func_call.__qualname__
+        + " failed. The "
+        + func_name
+        + " metric will not be recorded. Metric error: "
+        + str(err)
+    )
+    _logger.warning(msg)
+
+
+def _log_specialized_estimator_content(fitted_estimator, run_id, fit_args, fit_kwargs):
+    import sklearn
+
+    name_score_dict = {}
+    try:
+        if sklearn.base.is_classifier(fitted_estimator):
```

Comment:
Awesome! Can we add a test for the case when model prediction fails to ensure that we log a warning but don't throw an exception?

Reply:
Added a test:)